inc/managers/state_manager.hpp

```diff
@@ -25,10 +25,9 @@ class StateManager {
   StateManager();
   ~StateManager();
 
-  Sprite* sprite = new Sprite;

```

Comment:
It is better to allocate/deallocate in constructor

Reply:
Yeah! THe entyre code will be refactored xD